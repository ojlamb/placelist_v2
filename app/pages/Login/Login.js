import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as sessionActions from 'common/actions/session.actions';
import LoginView from './Login.view';

export const _mapStateToProps = state => ({
  session: state.session
});

// istanbul ignore next
const _mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(sessionActions, dispatch)
});

export default connect(_mapStateToProps, _mapDispatchToProps)(LoginView);
