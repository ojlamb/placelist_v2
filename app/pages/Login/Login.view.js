import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import FormControl from '@material-ui/core/FormControl';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import LockIcon from '@material-ui/icons/LockOutlined';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import withStyles from '@material-ui/core/styles/withStyles';

export const styles = theme => ({
  layout: {
    width: 'auto',
    display: 'block', // Fix IE11 issue.
    marginLeft: theme.spacing.unit * 3,
    marginRight: theme.spacing.unit * 3,
    [theme.breakpoints.up(400 + theme.spacing.unit * 3 * 2)]: {
      width: 400,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  },
  paper: {
    marginTop: theme.spacing.unit * 8,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: `${theme.spacing.unit * 2}px ${theme.spacing.unit * 3}px ${theme
      .spacing.unit * 3}px`
  },
  avatar: {
    margin: theme.spacing.unit,
    backgroundColor: theme.palette.secondary.main
  },
  form: {
    width: '100%', // Fix IE11 issue.
    marginTop: theme.spacing.unit
  },
  submit: {
    marginTop: theme.spacing.unit * 3
  },
  link: {
    textDecoration: 'none'
  },
  errorText: {
    textAlign: 'center',
    margin: 5
  }
});

export class LoginPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      credentials: { email: '', password: '' },
      saving: false,
      error: false,
      errorMessage: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onSave = this.onSave.bind(this);
  }

  onChange(event) {
    const field = event.target.name;
    const { credentials } = this.state;
    credentials[field] = event.target.value;
    return this.setState({
      ...this.statestate,
      credentials,
      error: false
    });
  }

  onSave(event) {
    event.preventDefault();
    this.setState({ saving: true });
    this.props.actions.loginUser(this.state.credentials).then(() => this.redirect())
      .catch((error) => {
        this.setState(prevState => ({
          ...prevState,
          saving: false,
          error: true,
          errorMessage: 'Email or Password Incorrect',
          serverError: error
        }));
      });
  }

  redirect() {
    this.props.history.push('/');
  }

  render() {
    const { classes } = this.props;
    const { error, errorMessage } = this.state;
    return (
      <React.Fragment>
        <CssBaseline />
        <main className={classes.layout}>
          <Paper className={classes.paper}>
            <Avatar className={classes.avatar}>
              <LockIcon />
            </Avatar>
            <Typography variant="h5">Sign in to PlaceList</Typography>
            <form className={classes.form}>
              <FormControl margin="normal" required fullWidth>
                <InputLabel htmlFor="email">Email Address</InputLabel>
                <Input
                  id="email"
                  name="email"
                  autoComplete="email"
                  autoFocus
                  value={this.state.credentials.email}
                  onChange={this.onChange}
                />
              </FormControl>
              <FormControl margin="normal" required fullWidth>
                <InputLabel htmlFor="password">Password</InputLabel>
                <Input
                  name="password"
                  type="password"
                  id="password"
                  value={this.state.credentials.password}
                  autoComplete="current-password"
                  onChange={this.onChange}
                />
              </FormControl>
              <Button
                type="submit"
                fullWidth
                variant="contained"
                color="primary"
                className={classes.submit}
                onClick={this.onSave}
              >
                Sign in
              </Button>
              {error &&
                <Typography variant="caption" color="error" className={classes.errorText}>{errorMessage}</Typography>
              }
            </form>
            <p>
              New User?
              <Link className={classes.link} to="/signup">
                <Button size="small" color="primary">
                  Sign Up
                </Button>
              </Link>
            </p>
          </Paper>
        </main>
      </React.Fragment>
    );
  }
}


LoginPage.propTypes = {
  actions: PropTypes.object.isRequired,
  classes: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
};

export default withStyles(styles)(LoginPage);
