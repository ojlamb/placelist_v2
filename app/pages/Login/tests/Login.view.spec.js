import React from 'react';
import theme from 'test/theme';
import { LoginPage, styles } from '../Login.view';

const getProps = actions => ({
  classes: {},
  actions,
  history: {
    push: jest.fn()
  }
});

const getJSX = props => (<LoginPage {...getProps(props)} />);

describe('SignupView', () => {
  describe('styles', () => {
    it('matches snapshot', () => {
      expect(styles(theme)).toMatchSnapshot();
    });
  });
  describe('snapshots', () => {
    const scenarios = [
      { loginUser: jest.fn().mockReturnValueOnce(Promise.resolve({ data: 'data' })) },
    ];

    scenarios.forEach((props, i) => {
      describe(`Scenario ${i + 1}`, () => {
        it('matches snapshot', () => {
          const wrapper = shallow(getJSX(props));
          expect(wrapper).toMatchSnapshot();
        });
      });
    });
  });
  describe('methods', () => {
    let wrapper;
    let instance;
    beforeAll(() => {
      const props = { loginUser: jest.fn().mockReturnValueOnce(Promise.resolve({ data: 'data' })) };
      wrapper = shallow(<LoginPage {...getProps(props)} />);
      instance = wrapper.instance();
    });
    it('updates the state for credentials', () => {
      instance.onChange({ target: { name: 'email', value: 'FOO' } });
      const credentials = wrapper.state('credentials');
      expect(credentials.email).toEqual('FOO');
    });
    it('updates the state when saving a user', () => {
      const login = jest.spyOn(instance, 'onSave');
      instance.onSave({ preventDefault: jest.fn() });
      expect(login).toHaveBeenCalled();
      expect(wrapper.state('saving')).toBe(true);
    });
  });
});
