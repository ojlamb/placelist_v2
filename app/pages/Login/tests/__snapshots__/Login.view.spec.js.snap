// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupView snapshots Scenario 1 matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginPage
    actions={
      Object {
        "loginUser": [MockFunction],
      }
    }
    classes={Object {}}
    history={
      Object {
        "push": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <WithStyles(CssBaseline) />,
        <main>
          <WithStyles(Paper)>
            <WithStyles(Avatar)>
              <pure(LockOutlinedIcon) />
            </WithStyles(Avatar)>
            <WithStyles(Typography)
              variant="h5"
            >
              Sign in to PlaceList
            </WithStyles(Typography)>
            <form>
              <WithStyles(FormControl)
                fullWidth={true}
                margin="normal"
                required={true}
              >
                <WithStyles(WithFormControlContext(InputLabel))
                  htmlFor="email"
                >
                  Email Address
                </WithStyles(WithFormControlContext(InputLabel))>
                <WithStyles(Input)
                  autoComplete="email"
                  autoFocus={true}
                  id="email"
                  name="email"
                  onChange={[Function]}
                  value=""
                />
              </WithStyles(FormControl)>
              <WithStyles(FormControl)
                fullWidth={true}
                margin="normal"
                required={true}
              >
                <WithStyles(WithFormControlContext(InputLabel))
                  htmlFor="password"
                >
                  Password
                </WithStyles(WithFormControlContext(InputLabel))>
                <WithStyles(Input)
                  autoComplete="current-password"
                  id="password"
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value=""
                />
              </WithStyles(FormControl)>
              <WithStyles(Button)
                color="primary"
                fullWidth={true}
                onClick={[Function]}
                type="submit"
                variant="contained"
              >
                Sign in
              </WithStyles(Button)>
            </form>
            <p>
              New User?
              <Link
                replace={false}
                to="/signup"
              >
                <WithStyles(Button)
                  color="primary"
                  size="small"
                >
                  Sign Up
                </WithStyles(Button)>
              </Link>
            </p>
          </WithStyles(Paper)>
        </main>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(Paper)>
            <WithStyles(Avatar)>
              <pure(LockOutlinedIcon) />
            </WithStyles(Avatar)>
            <WithStyles(Typography)
              variant="h5"
            >
              Sign in to PlaceList
            </WithStyles(Typography)>
            <form>
              <WithStyles(FormControl)
                fullWidth={true}
                margin="normal"
                required={true}
              >
                <WithStyles(WithFormControlContext(InputLabel))
                  htmlFor="email"
                >
                  Email Address
                </WithStyles(WithFormControlContext(InputLabel))>
                <WithStyles(Input)
                  autoComplete="email"
                  autoFocus={true}
                  id="email"
                  name="email"
                  onChange={[Function]}
                  value=""
                />
              </WithStyles(FormControl)>
              <WithStyles(FormControl)
                fullWidth={true}
                margin="normal"
                required={true}
              >
                <WithStyles(WithFormControlContext(InputLabel))
                  htmlFor="password"
                >
                  Password
                </WithStyles(WithFormControlContext(InputLabel))>
                <WithStyles(Input)
                  autoComplete="current-password"
                  id="password"
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value=""
                />
              </WithStyles(FormControl)>
              <WithStyles(Button)
                color="primary"
                fullWidth={true}
                onClick={[Function]}
                type="submit"
                variant="contained"
              >
                Sign in
              </WithStyles(Button)>
            </form>
            <p>
              New User?
              <Link
                replace={false}
                to="/signup"
              >
                <WithStyles(Button)
                  color="primary"
                  size="small"
                >
                  Sign Up
                </WithStyles(Button)>
              </Link>
            </p>
          </WithStyles(Paper)>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Avatar)>
                <pure(LockOutlinedIcon) />
              </WithStyles(Avatar)>,
              <WithStyles(Typography)
                variant="h5"
              >
                Sign in to PlaceList
              </WithStyles(Typography)>,
              <form>
                <WithStyles(FormControl)
                  fullWidth={true}
                  margin="normal"
                  required={true}
                >
                  <WithStyles(WithFormControlContext(InputLabel))
                    htmlFor="email"
                  >
                    Email Address
                  </WithStyles(WithFormControlContext(InputLabel))>
                  <WithStyles(Input)
                    autoComplete="email"
                    autoFocus={true}
                    id="email"
                    name="email"
                    onChange={[Function]}
                    value=""
                  />
                </WithStyles(FormControl)>
                <WithStyles(FormControl)
                  fullWidth={true}
                  margin="normal"
                  required={true}
                >
                  <WithStyles(WithFormControlContext(InputLabel))
                    htmlFor="password"
                  >
                    Password
                  </WithStyles(WithFormControlContext(InputLabel))>
                  <WithStyles(Input)
                    autoComplete="current-password"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                </WithStyles(FormControl)>
                <WithStyles(Button)
                  color="primary"
                  fullWidth={true}
                  onClick={[Function]}
                  type="submit"
                  variant="contained"
                >
                  Sign in
                </WithStyles(Button)>
              </form>,
              <p>
                New User?
                <Link
                  replace={false}
                  to="/signup"
                >
                  <WithStyles(Button)
                    color="primary"
                    size="small"
                  >
                    Sign Up
                  </WithStyles(Button)>
                </Link>
              </p>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <pure(LockOutlinedIcon) />,
                "className": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Sign in to PlaceList",
                "variant": "h5",
              },
              "ref": null,
              "rendered": "Sign in to PlaceList",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <WithStyles(FormControl)
                    fullWidth={true}
                    margin="normal"
                    required={true}
                  >
                    <WithStyles(WithFormControlContext(InputLabel))
                      htmlFor="email"
                    >
                      Email Address
                    </WithStyles(WithFormControlContext(InputLabel))>
                    <WithStyles(Input)
                      autoComplete="email"
                      autoFocus={true}
                      id="email"
                      name="email"
                      onChange={[Function]}
                      value=""
                    />
                  </WithStyles(FormControl)>,
                  <WithStyles(FormControl)
                    fullWidth={true}
                    margin="normal"
                    required={true}
                  >
                    <WithStyles(WithFormControlContext(InputLabel))
                      htmlFor="password"
                    >
                      Password
                    </WithStyles(WithFormControlContext(InputLabel))>
                    <WithStyles(Input)
                      autoComplete="current-password"
                      id="password"
                      name="password"
                      onChange={[Function]}
                      type="password"
                      value=""
                    />
                  </WithStyles(FormControl)>,
                  <WithStyles(Button)
                    color="primary"
                    fullWidth={true}
                    onClick={[Function]}
                    type="submit"
                    variant="contained"
                  >
                    Sign in
                  </WithStyles(Button)>,
                  false,
                ],
                "className": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <WithStyles(WithFormControlContext(InputLabel))
                        htmlFor="email"
                      >
                        Email Address
                      </WithStyles(WithFormControlContext(InputLabel))>,
                      <WithStyles(Input)
                        autoComplete="email"
                        autoFocus={true}
                        id="email"
                        name="email"
                        onChange={[Function]}
                        value=""
                      />,
                    ],
                    "fullWidth": true,
                    "margin": "normal",
                    "required": true,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Email Address",
                        "htmlFor": "email",
                      },
                      "ref": null,
                      "rendered": "Email Address",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "autoComplete": "email",
                        "autoFocus": true,
                        "id": "email",
                        "name": "email",
                        "onChange": [Function],
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <WithStyles(WithFormControlContext(InputLabel))
                        htmlFor="password"
                      >
                        Password
                      </WithStyles(WithFormControlContext(InputLabel))>,
                      <WithStyles(Input)
                        autoComplete="current-password"
                        id="password"
                        name="password"
                        onChange={[Function]}
                        type="password"
                        value=""
                      />,
                    ],
                    "fullWidth": true,
                    "margin": "normal",
                    "required": true,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Password",
                        "htmlFor": "password",
                      },
                      "ref": null,
                      "rendered": "Password",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "autoComplete": "current-password",
                        "id": "password",
                        "name": "password",
                        "onChange": [Function],
                        "type": "password",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sign in",
                    "className": undefined,
                    "color": "primary",
                    "fullWidth": true,
                    "onClick": [Function],
                    "type": "submit",
                    "variant": "contained",
                  },
                  "ref": null,
                  "rendered": "Sign in",
                  "type": [Function],
                },
                false,
              ],
              "type": "form",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "New User?",
                  <Link
                    replace={false}
                    to="/signup"
                  >
                    <WithStyles(Button)
                      color="primary"
                      size="small"
                    >
                      Sign Up
                    </WithStyles(Button)>
                  </Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "New User?",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(Button)
                      color="primary"
                      size="small"
                    >
                      Sign Up
                    </WithStyles(Button)>,
                    "className": undefined,
                    "replace": false,
                    "to": "/signup",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Sign Up",
                      "color": "primary",
                      "size": "small",
                    },
                    "ref": null,
                    "rendered": "Sign Up",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": "p",
            },
          ],
          "type": [Function],
        },
        "type": "main",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <WithStyles(CssBaseline) />,
          <main>
            <WithStyles(Paper)>
              <WithStyles(Avatar)>
                <pure(LockOutlinedIcon) />
              </WithStyles(Avatar)>
              <WithStyles(Typography)
                variant="h5"
              >
                Sign in to PlaceList
              </WithStyles(Typography)>
              <form>
                <WithStyles(FormControl)
                  fullWidth={true}
                  margin="normal"
                  required={true}
                >
                  <WithStyles(WithFormControlContext(InputLabel))
                    htmlFor="email"
                  >
                    Email Address
                  </WithStyles(WithFormControlContext(InputLabel))>
                  <WithStyles(Input)
                    autoComplete="email"
                    autoFocus={true}
                    id="email"
                    name="email"
                    onChange={[Function]}
                    value=""
                  />
                </WithStyles(FormControl)>
                <WithStyles(FormControl)
                  fullWidth={true}
                  margin="normal"
                  required={true}
                >
                  <WithStyles(WithFormControlContext(InputLabel))
                    htmlFor="password"
                  >
                    Password
                  </WithStyles(WithFormControlContext(InputLabel))>
                  <WithStyles(Input)
                    autoComplete="current-password"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                </WithStyles(FormControl)>
                <WithStyles(Button)
                  color="primary"
                  fullWidth={true}
                  onClick={[Function]}
                  type="submit"
                  variant="contained"
                >
                  Sign in
                </WithStyles(Button)>
              </form>
              <p>
                New User?
                <Link
                  replace={false}
                  to="/signup"
                >
                  <WithStyles(Button)
                    color="primary"
                    size="small"
                  >
                    Sign Up
                  </WithStyles(Button)>
                </Link>
              </p>
            </WithStyles(Paper)>
          </main>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(Paper)>
              <WithStyles(Avatar)>
                <pure(LockOutlinedIcon) />
              </WithStyles(Avatar)>
              <WithStyles(Typography)
                variant="h5"
              >
                Sign in to PlaceList
              </WithStyles(Typography)>
              <form>
                <WithStyles(FormControl)
                  fullWidth={true}
                  margin="normal"
                  required={true}
                >
                  <WithStyles(WithFormControlContext(InputLabel))
                    htmlFor="email"
                  >
                    Email Address
                  </WithStyles(WithFormControlContext(InputLabel))>
                  <WithStyles(Input)
                    autoComplete="email"
                    autoFocus={true}
                    id="email"
                    name="email"
                    onChange={[Function]}
                    value=""
                  />
                </WithStyles(FormControl)>
                <WithStyles(FormControl)
                  fullWidth={true}
                  margin="normal"
                  required={true}
                >
                  <WithStyles(WithFormControlContext(InputLabel))
                    htmlFor="password"
                  >
                    Password
                  </WithStyles(WithFormControlContext(InputLabel))>
                  <WithStyles(Input)
                    autoComplete="current-password"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                </WithStyles(FormControl)>
                <WithStyles(Button)
                  color="primary"
                  fullWidth={true}
                  onClick={[Function]}
                  type="submit"
                  variant="contained"
                >
                  Sign in
                </WithStyles(Button)>
              </form>
              <p>
                New User?
                <Link
                  replace={false}
                  to="/signup"
                >
                  <WithStyles(Button)
                    color="primary"
                    size="small"
                  >
                    Sign Up
                  </WithStyles(Button)>
                </Link>
              </p>
            </WithStyles(Paper)>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Avatar)>
                  <pure(LockOutlinedIcon) />
                </WithStyles(Avatar)>,
                <WithStyles(Typography)
                  variant="h5"
                >
                  Sign in to PlaceList
                </WithStyles(Typography)>,
                <form>
                  <WithStyles(FormControl)
                    fullWidth={true}
                    margin="normal"
                    required={true}
                  >
                    <WithStyles(WithFormControlContext(InputLabel))
                      htmlFor="email"
                    >
                      Email Address
                    </WithStyles(WithFormControlContext(InputLabel))>
                    <WithStyles(Input)
                      autoComplete="email"
                      autoFocus={true}
                      id="email"
                      name="email"
                      onChange={[Function]}
                      value=""
                    />
                  </WithStyles(FormControl)>
                  <WithStyles(FormControl)
                    fullWidth={true}
                    margin="normal"
                    required={true}
                  >
                    <WithStyles(WithFormControlContext(InputLabel))
                      htmlFor="password"
                    >
                      Password
                    </WithStyles(WithFormControlContext(InputLabel))>
                    <WithStyles(Input)
                      autoComplete="current-password"
                      id="password"
                      name="password"
                      onChange={[Function]}
                      type="password"
                      value=""
                    />
                  </WithStyles(FormControl)>
                  <WithStyles(Button)
                    color="primary"
                    fullWidth={true}
                    onClick={[Function]}
                    type="submit"
                    variant="contained"
                  >
                    Sign in
                  </WithStyles(Button)>
                </form>,
                <p>
                  New User?
                  <Link
                    replace={false}
                    to="/signup"
                  >
                    <WithStyles(Button)
                      color="primary"
                      size="small"
                    >
                      Sign Up
                    </WithStyles(Button)>
                  </Link>
                </p>,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <pure(LockOutlinedIcon) />,
                  "className": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Sign in to PlaceList",
                  "variant": "h5",
                },
                "ref": null,
                "rendered": "Sign in to PlaceList",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <WithStyles(FormControl)
                      fullWidth={true}
                      margin="normal"
                      required={true}
                    >
                      <WithStyles(WithFormControlContext(InputLabel))
                        htmlFor="email"
                      >
                        Email Address
                      </WithStyles(WithFormControlContext(InputLabel))>
                      <WithStyles(Input)
                        autoComplete="email"
                        autoFocus={true}
                        id="email"
                        name="email"
                        onChange={[Function]}
                        value=""
                      />
                    </WithStyles(FormControl)>,
                    <WithStyles(FormControl)
                      fullWidth={true}
                      margin="normal"
                      required={true}
                    >
                      <WithStyles(WithFormControlContext(InputLabel))
                        htmlFor="password"
                      >
                        Password
                      </WithStyles(WithFormControlContext(InputLabel))>
                      <WithStyles(Input)
                        autoComplete="current-password"
                        id="password"
                        name="password"
                        onChange={[Function]}
                        type="password"
                        value=""
                      />
                    </WithStyles(FormControl)>,
                    <WithStyles(Button)
                      color="primary"
                      fullWidth={true}
                      onClick={[Function]}
                      type="submit"
                      variant="contained"
                    >
                      Sign in
                    </WithStyles(Button)>,
                    false,
                  ],
                  "className": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        <WithStyles(WithFormControlContext(InputLabel))
                          htmlFor="email"
                        >
                          Email Address
                        </WithStyles(WithFormControlContext(InputLabel))>,
                        <WithStyles(Input)
                          autoComplete="email"
                          autoFocus={true}
                          id="email"
                          name="email"
                          onChange={[Function]}
                          value=""
                        />,
                      ],
                      "fullWidth": true,
                      "margin": "normal",
                      "required": true,
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Email Address",
                          "htmlFor": "email",
                        },
                        "ref": null,
                        "rendered": "Email Address",
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "autoComplete": "email",
                          "autoFocus": true,
                          "id": "email",
                          "name": "email",
                          "onChange": [Function],
                          "value": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        <WithStyles(WithFormControlContext(InputLabel))
                          htmlFor="password"
                        >
                          Password
                        </WithStyles(WithFormControlContext(InputLabel))>,
                        <WithStyles(Input)
                          autoComplete="current-password"
                          id="password"
                          name="password"
                          onChange={[Function]}
                          type="password"
                          value=""
                        />,
                      ],
                      "fullWidth": true,
                      "margin": "normal",
                      "required": true,
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Password",
                          "htmlFor": "password",
                        },
                        "ref": null,
                        "rendered": "Password",
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "autoComplete": "current-password",
                          "id": "password",
                          "name": "password",
                          "onChange": [Function],
                          "type": "password",
                          "value": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Sign in",
                      "className": undefined,
                      "color": "primary",
                      "fullWidth": true,
                      "onClick": [Function],
                      "type": "submit",
                      "variant": "contained",
                    },
                    "ref": null,
                    "rendered": "Sign in",
                    "type": [Function],
                  },
                  false,
                ],
                "type": "form",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "New User?",
                    <Link
                      replace={false}
                      to="/signup"
                    >
                      <WithStyles(Button)
                        color="primary"
                        size="small"
                      >
                        Sign Up
                      </WithStyles(Button)>
                    </Link>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "New User?",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <WithStyles(Button)
                        color="primary"
                        size="small"
                      >
                        Sign Up
                      </WithStyles(Button)>,
                      "className": undefined,
                      "replace": false,
                      "to": "/signup",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Sign Up",
                        "color": "primary",
                        "size": "small",
                      },
                      "ref": null,
                      "rendered": "Sign Up",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": "p",
              },
            ],
            "type": [Function],
          },
          "type": "main",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`SignupView styles matches snapshot 1`] = `
Object {
  "avatar": Object {
    "backgroundColor": "#000",
    "margin": 10,
  },
  "errorText": Object {
    "margin": 5,
    "textAlign": "center",
  },
  "form": Object {
    "marginTop": 10,
    "width": "100%",
  },
  "layout": Object {
    "display": "block",
    "marginLeft": 30,
    "marginRight": 30,
    "undefined": Object {
      "marginLeft": "auto",
      "marginRight": "auto",
      "width": 400,
    },
    "width": "auto",
  },
  "link": Object {
    "textDecoration": "none",
  },
  "paper": Object {
    "alignItems": "center",
    "display": "flex",
    "flexDirection": "column",
    "marginTop": 80,
    "padding": "20px 30px 30px",
  },
  "submit": Object {
    "marginTop": 30,
  },
}
`;
